src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,111 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.TodoCommand;
+import duke.command.UnknownCommand;
+import duke.command.UnmarkCommand;
+
+public class Parser {
+
+    /**
+     * Checks if the command line input is a proper argument.
+     *
+     * @param arr String array that contains the seperate parts of the command line argument.
+     * @return True if the command line input is of the correct form.
+     */
+    private static boolean inputChecker(String[] arr) {
+        if (arr.length  < 2) {
+            return false;
+        }
+        if (arr[1].isBlank()) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Converts the command line input to call the command.
+     *
+     * @param input The commmand line input.
+     * @return Command with respect ot the command line input.
+     */
+    public static Command parse (String input) throws DukeException{
+        String[] arr = input.split(" ", 2);
+
+        if (input.equals("bye")) {
+            return new ExitCommand();
+        }
+
+        else if (input.equals("list")) {
+            return new ListCommand();
+        }
+
+        else if (arr[0].equals("mark")){
+            if (!inputChecker(arr)) {
+                throw DukeException.MarkIndexEmptyException();
+            } else {
+                return new MarkCommand(arr);
+            }
+
+        }
+        else if (arr[0].equals("unmark")) {
```

Comment:
I kind of understand why the if else blocks were formatted this way because of how many else blocks there are though this is still a violation of the coding standard as shown below. Perhaps a switch block would be better here?

```java
if (condition) {
    statements;
} else if (condition) {
    statements;
} else {
    statements;
}
```